go/tools/builders/filter.go

```diff
@@ -19,64 +19,77 @@ import (
 	"go/build"
 	"go/parser"
 	"go/token"
-	"io/ioutil"
 	"log"
 	"path/filepath"
 	"strconv"
 	"strings"
 )
 
+type goMetadata struct {
+	Filename string
```

Comment:
Nit: make these private.

Doesn't make any difference since this is `main`, but it's better to default to private.